packages/Webkul/Sales/src/Repositories/OrderRepository.php

```diff
@@ -386,6 +388,113 @@ public function collectTotals($order)
+        }
+
+        return $this->count();
+    }
+
+    /**
+     * Calculate sale amount by date.
+     */
+    public function calculateSaleAmountByDate(?Carbon $from = null, ?Carbon $to = null): ?float
+    {
+        if ($from && $to) {
+            return $this->getModel()
+                ->whereBetween('created_at', [$from, $to])
+                ->sum(DB::raw('base_grand_total_invoiced - base_grand_total_refunded'));
+        }
+
+        if ($from) {
+            return $this->getModel()
+                ->where('created_at', '>=', $from)
+                ->sum(DB::raw('base_grand_total_invoiced - base_grand_total_refunded'));
+        }
+
+        if ($to) {
+            return $this->getModel()
+                ->where('created_at', '<=', $to)
+                ->sum(DB::raw('base_grand_total_invoiced - base_grand_total_refunded'));
+        }
+
+        return $this->model->sum(DB::raw('base_grand_total_invoiced - base_grand_total_refunded'));
+    }
+
+    /**
+     * Calculate average sale amount by date.
+     */
+    public function calculateAvgSaleAmountByDate(?Carbon $from = null, ?Carbon $to = null): ?float
+    {
+        if ($from && $to) {
+            return $this->getModel()
+                ->whereBetween('created_at', [$from, $to])
+                ->avg(DB::raw('base_grand_total_invoiced - base_grand_total_refunded'));
+        }
+
+        if ($from) {
+            return $this->getModel()
+                ->where('created_at', '>=', $from)
+                ->avg(DB::raw('base_grand_total_invoiced - base_grand_total_refunded'));
+        }
+
+        if ($to) {
+            return $this->getModel()
+                ->where('created_at', '<=', $to)
+                ->avg(DB::raw('base_grand_total_invoiced - base_grand_total_refunded'));
+        }
+
+        return $this->getModel()->avg(DB::raw('base_grand_total_invoiced - base_grand_total_refunded'));
+    }
+
+    /**
+     * Get customer with most sales by date.
+     */
+    public function getCustomerWithMostSalesByDate(?Carbon $from = null, ?Carbon $to = null): Collection
+    {
+        $dbPrefix = DB::getTablePrefix();
+
+        $query = $this->getModel()
+            ->whereNotIn("{$dbPrefix}orders.status", ['closed', 'canceled'])
+            ->select(
+                'customer_id',
+                'customer_email',
+                'customer_first_name',
+                'customer_last_name',
+                DB::raw("(
+                    SUM(base_grand_total) -
+                    SUM(
+                        IFNULL(
+                            (SELECT SUM(base_grand_total) FROM {$dbPrefix}refunds WHERE {$dbPrefix}refunds.order_id = {$dbPrefix}orders.id),
+                            0)
+                        )
+                    ) as total_base_grand_total")
+            );
+
+        if ($from && $to) {
+            $query->whereBetween("{$dbPrefix}orders.created_at", [$from, $to]);
```

Comment:
No need for a table prefix here. Laravel automcatically handles this one. Only raw queries are needed. You can revert this one.

Reply:
Yes, you're correct.